render.c

```diff
@@ -41,30 +42,40 @@ void render_frame_background(struct swaylock_surface *surface) {
 		return; // not yet configured
 	}
 
-	struct pool_buffer *buffer = get_next_buffer(state->shm,
-			surface->buffers, buffer_width, buffer_height);
-	if (buffer == NULL) {
-		return;
-	}
+	if (buffer_width != surface->last_buffer_width ||
+		buffer_height != surface->last_buffer_height) {
```

Comment:
Style nit: two tabs for `if `continuation lines, to be able to tell apart the condition and the body